src/platforms/common/aux_serial.c

```diff
@@ -87,6 +87,23 @@ static char aux_serial_transmit_buffer[AUX_UART_BUFFER_SIZE];
 #define usart_set_parity(uart, parity)     uart_set_parity(uart, parity)
 #endif
 
+void bmp_usart_set_baudrate(uint32_t usart, uint32_t baud_rate)
+{
+#if defined(USART_CR1_OVER8)
+	const uint32_t clock = rcc_get_usart_clk_freq(usart);
+	/* Autoselect default OVER16 unless a very high baudrate requested */
+	if (baud_rate <= clock / 16U) {
```

Comment:
This could be fine, but please check if a BRR of 1 is allowed. Otherwise this will need to check for `clock / 32U` or possibly even higher to check for a large enough dividend to work.

Reply:
RM0383 tells the highest legal baudrate is 10.5Mbaud for 84MHz Pclk in OVER8 when a fixed-point number 1.000 is "programmed in the baud rate register" (Table 82, end of example baudrate tables).
I understand the refman such that DIV_Mantissa has to be at least 1, and DIV_Fraction can be anything. So I limit to /16 normally, Q12.4 = 1+0/16, and to /8 specially, Q12.3 = 1+0/8. It corresponds to BRR of 16 (0x10) and 16 (0x10), not 1 (0x01). This is an improvement to previous behaviour where BMF would silently set any BRR, including illegal values (1..7, i.e. 0+1/16 .. 0+7/16 etc.)

Also see 19.3.3 Receiver, Selecting the proper oversampling method, 
> Depending on the application:
•select oversampling by 8 (OVER8=1) to achieve higher speed (up to fPCLK/8). ...
•select oversampling by 16 (OVER8=0) to increase the tolerance of the receiver to clock deviations. In this case, the maximum speed is limited to maximum fPCLK/16.

Reply:
Long version short: Because of matching code in libopencm3 which generates a BRR of 0x0010U when `baud_rate == clock / 16U` it's a sufficient check? (trying to summarise what you're saying)

If that is the case then we're happy and can mark this review note respolved.

Reply:
Expanded check to take care of four ranges. Added in-range check for F103.